### PR TITLE
Fixes #24480 - fix org switching on subs page

### PR DESCRIPTION
**Pull request by amirfefer**
Certain flows will allow the user to see subs across any organization. 
This PR fixes it -  when clicking on the `Any Organization` ,  a select organization page should appear.

### CODE REVIEW

File: webpack/components/WithOrganization/withOrganization.js [hunk 1]
```diff
@@ -1,10 +1,25 @@
...
 function withOrganization(WrappedComponent, redirectPath) {
-  return class CheckOrg extends Component {
+  const checkOrg = class CheckOrg extends Component {
+    constructor(props) {
+      super(props);
+      this.state = { anyOrg: null, currentOrg: null };
+    }
+    static getDerivedStateFromProps(newProps) {
+      const uiOrgName = document.getElementById('organization-dropdown').children[0].text.trim();
+      if (uiOrgName === 'Any Organization' && !newProps.location.state) {
```
Comment (waldenraines) [inline webpack/components/WithOrganization/withOrganization.js]: This seems a little fragile, can we just check for the lack of an org id?

**Comment by amirfefer** [comment]
@waldenraines  I've added some tests

**Comment by waldenraines** [comment]
@amirfefer this works well and thanks for adding tests but this now needs a rebase.

File: webpack/containers/Application/config.js [hunk 2]
```diff
@@ -4,32 +4,34 @@ import UpstreamSubscriptions from '../../scenes/Subscriptions/UpstreamSubscripti
...
+import withHeader from './withHeaders';
 
 // eslint-disable-next-line import/prefer-default-export
 export const links = [
   {
-    text: __('RH Repos'),
     path: 'redhat_repositories',
-    component: WithOrganization(Repos, '/redhat_repositories'),
+    component: WithOrganization(
+      withHeader(Repos, { title: __('RH Repos') }),
```
Comment (amirfefer) [inline webpack/containers/Application/config.js]: The HOC's order is importent

File: webpack/containers/Application/Routes.js [hunk 3]
```diff
@@ -1,22 +1,11 @@
 import React from 'react';
 import { Route } from 'react-router-dom';
 import { links } from './config';
-import Header from './Headers';
 
 export default () => (
   <div>
-    {links.map(({ path, component, text }) => {
-      const Page = component;
-      const withHeader = () => (
```
Comment (amirfefer) [inline webpack/containers/Application/Routes.js]: on every route change `withHeader`  will recreate, it should exist as a separate component.

**Comment by amirfefer** [comment]
@waldenraines  I've rebased and fix some issues, would you mind to have a quick look?

File: package.json [hunk 4]
```diff
@@ -31,8 +31,8 @@
     "babel-polyfill": "^6.26.0",
     "babel-preset-env": "^1.6.0",
     "babel-preset-react": "^6.24.1",
-    "enzyme": "^3.2.0",
-    "enzyme-adapter-react-16": "^1.1.0",
+    "enzyme": "^3.4.0",
+    "enzyme-adapter-react-16.3": "^1.0.0",
```
Comment (waldenraines) [inline package.json]: Are these changes required for this?  If so we need to add a request to update these packages downstream.

Comment (amirfefer) [reply package.json]: works locally without these.

**Comment by waldenraines** [comment]
@amirfefer sorry, I created some more conflicts for you here by merging #7630

**Comment by waldenraines** [comment]
> There were the following issues with the commit message:

Dang, I was hoping I could fix these conflicts for you and merge but apparently not.

**Comment by amirfefer** [comment]
@waldenraines  - rebased :)

**Comment by waldenraines** [comment]
[test katello]
